.github/workflows/lint-check-formatting-and-test.yaml

```diff
@@ -0,0 +1,47 @@
+# Adapted from  GH documentation: https://docs.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Lint, check formatting and run tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```

Comment:
```suggestion
```

This is possibly redundant and not wante?

Reply:
Is it? Ah OK, it looked like weird YAML but I'll change to:

```yaml
on:
  push:
    branches:
      - master
  pull_request:
```

Reply:
Changed ✅